tests/e2e/defaultvmgroup_test.go

```diff
@@ -57,38 +65,111 @@ var _ = Suite(&DefaultVMGroupTestSuite{})
+
+	// Verify if volume exists in default vmgroup
+	isVolInVmgroup := admincli.CheckVolumeExistInVmgroup(s.config.EsxHost, s.volumeNames[0], admincliconst.DefaultVMgroup)
+	c.Assert(isVolInVmgroup, Equals, true, Commentf("Volume [%s] does not belong to vmgroup [%s]", s.volumeNames[0], admincliconst.DefaultVMgroup))
+
+	// Delete default vmgroup and verify it does not exist
+	admincli.DeleteVMgroup(s.config.EsxHost, admincliconst.DefaultVMgroup)
+	isVmgroupAvailable := admincli.CheckVmgroupAvailability(s.config.EsxHost, admincliconst.DefaultVMgroup)
+	c.Assert(isVmgroupAvailable, Equals, false, Commentf("Falied to delete vmgroup %s .", admincliconst.DefaultVMgroup))
+
+	// Create a volume - operation should fail as default vmgroup no longer exist
+	out, err := dockercli.CreateVolume(s.config.DockerHosts[0], s.volumeNames[1])
+	c.Assert(err, Not(IsNil), Commentf(out))
+
+	// Verify existing volume - volumeName1 does not belong to any vmgroup (N/A)
+	isVolInVmgroup = admincli.CheckVolumeExistInVmgroup(s.config.EsxHost, s.volumeNames[0], NotAvailable)
+	c.Assert(isVolInVmgroup, Equals, true, Commentf("Unexpected Behavior: Vmgroup for volume [%s] is not N/A. ", s.volumeNames[0]))
+
+	s.createDefaultVmgroup(c)
+
+	// Create a volume - operation should succeed as default vmgroup exists now
+	s.createVolume(c, s.config.DockerHosts[0], s.volumeNames[1])
+
+	// Check if volume was successfully created
+	isAvailable = verification.CheckVolumeAvailability(s.config.DockerHosts[0], s.volumeNames[1])
+	c.Assert(isAvailable, Equals, true, Commentf("Volume %s is not available after creation", s.volumeNames[1]))
+
+	// Verify if volume exists in default vmgroup
+	isVolInVmgroup = admincli.CheckVolumeExistInVmgroup(s.config.EsxHost, s.volumeNames[1], admincliconst.DefaultVMgroup)
+	c.Assert(isVolInVmgroup, Equals, true, Commentf("Volume [%s] does not belong to vmgroup [%s]", s.volumeNames[1], admincliconst.DefaultVMgroup))
+
+	out, err = ssh.InvokeCommand(s.config.DockerHosts[0], dockerconst.RemoveVolume+strings.Join(s.volumeNames, " "))
+	c.Assert(err, IsNil, Commentf(out))
+
+	misc.LogTestEnd(c.TestName())
+}
+
+// Create the default Vmgroup
+func (s *DefaultVMGroupTestSuite) createDefaultVmgroup(c *C) {
+	cmd := admincliconst.CreateVMgroup + admincliconst.DefaultVMgroup + " --default-datastore " + admincliconst.VMHomeDatastore
+	log.Printf("Creating default vmgroup %s", admincliconst.DefaultVMgroup)
+	ssh.InvokeCommand(s.config.EsxHost, cmd)
+	ssh.InvokeCommand(s.config.EsxHost, admincliconst.AddDatastoreToVMgroup+admincliconst.DefaultVMgroup+" --datastore=_ALL_DS --allow-create ")
+
+	// Verify default vmgroup exists
+	isVmgroupAvailable := admincli.CheckVmgroupAvailability(s.config.EsxHost, admincliconst.DefaultVMgroup)
+	c.Assert(isVmgroupAvailable, Equals, true, Commentf("Falied to create vmgroup %s .", admincliconst.DefaultVMgroup))
+}
+
+// Create volumes
+func (s *DefaultVMGroupTestSuite) createVolume(c *C, hostIp, volumeName string) {
```

Comment:
I'd suggest avoid passing *C to private functions. A private function should just take care of one specific task (for code reuse or whatever purpose), and return a result (e.g. string, boolean, error, etc.) that can be verified by the main test workflow using c.Assert().